cmake/LinkCudaLibraries.cmake

```diff
@@ -0,0 +1,33 @@
+# Link CUDA libraries to the given target, i.e.: `target_link_libraries(target <args>)`
```

Comment:
Can this be moved to `Dependencies.cmake`? Or perhaps extract parts of Dependencies.cmake in CUDA.cmake and keep it there?

Reply:
What is the benefit? This follows the common CMake convention that a function is defined in a file of the same name.  
And I'm not 100% sure `Dependencies.cmake` is included before each of the use points. Simply including the file prior to using it looks the simplest to me. But of course can move if that better suits the project conventions.

> Or perhaps extract parts of Dependencies.cmake in CUDA.cmake and keep it there?

Not sure I understand this exactly. Which parts should be extracted? And where would that new CUDA.cmake be included? By `Dependencies.cmake`?